pkg/cfn/manager/waiters.go

```diff
@@ -106,8 +107,45 @@ func (c *StackCollection) waitWithAcceptors(name string, acceptors []request.Wai
 	return nil
 }
 
-func (c *StackCollection) doWaitUntilStackIsCreated(name string) error {
-	return c.waitWithAcceptors(name,
+func (c *StackCollection) troubleshootStackFailureCause(i *Stack, desiredStatus string) {
```

Comment:
@richardcase do you know what the logic should be for the GPU case? Is it the AMI subscription issue that we need to address here? It's possible the code here already addresses that issues, as it would bubble up as node creation failure... I recall you also mentioned timeout, I don't think that is something we can connect to this code path directly, but I maybe wrong.

Reply:
I think it will bubble up as a node creation failure and the error message returned would probably be sufficient as it will be out here.

I'll see if i can test it out.